kernel/kernel-api/src/main/java/io/delta/kernel/internal/TransactionImpl.java

```diff
@@ -444,6 +445,10 @@ private TransactionCommitResult doCommit(
       // Action counters may be partially incremented from previous tries, reset the counters to 0
       transactionMetrics.resetActionCounters();
 
+      Optional<FileSizeHistogram> fileSizeHistogram =
+          isNewTable
+              ? Optional.of(FileSizeHistogram.createDefaultHistogram())
+              : readSnapshot.getCurrentCrcInfo().flatMap(CRCInfo::getFileSizeHistogram);
```

Comment:
Add a comment in what conditions we update the file size historing: new table, or table that already has filesize histogram available through CRC